### ISSUE TITLE
make check produced one error on Fedora Core 32 (FC 32) (Rawhide)

### ISSUE THREAD

**Issue by grgoffe** [curl/curl#4631]
### I did this

 get the latest curl from the repository and build. I had to disable-werror to get the build to continue. see deprecated messages below.  "make check" failed one test (see enclosed build log)

### I expected the following

clean and successful build

### curl/libcurl version

curl 7.68.0-DEV (x86_64-pc-linux-gnu) libcurl/7.68.0-DEV OpenSSL/1.1.1d-fips zlib/1.2.11
Release-Date: [unreleased]
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: alt-svc AsynchDNS Debug HTTPS-proxy IPv6 Largefile libz NTLM NTLM_WB SSL TLS-SRP TrackMemory UnixSockets

### operating system

Fedora Core 32 x86_64 (Rawhide)

Additional info:
~~~
vtls/openssl.c: In function ‘ossl_connect_step1’:
vtls/openssl.c:2467:5: error: ‘SSLv3_client_method’ is deprecated [-Werror=deprecated-declarations]
2467 |     req_method = SSLv3_client_method();
|     ^~~~~~~~~~
In file included from /usr/include/openssl/opensslconf.h:42,
from /usr/include/openssl/e_os2.h:13,
from /usr/include/openssl/ssl.h:15,
from vtls/openssl.c:49:
/usr/include/openssl/ssl.h:1869:1: note: declared here
1869 | DEPRECATEDIN_1_1_0(__owur const SSL_METHOD *SSLv3_client_method(void))
| ^~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:2480: vtls/libcurl_la-openssl.lo] Error 1
make[2]: Leaving directory '/export/home/tools/curl/curl/lib'
~~~

Output of "make check":
~~~
5288 test 2080...OK (1261 out of 1264, remaining: 00:06, took 2.0s, duration: 43:05)
5289 Warning: dict server unexpectedly alive
5290 TESTDONE: 1127 tests out of 1128 reported OK: 99%
5291 TESTFAIL: These test cases failed: 1509 
5292 TESTDONE: 1282 tests were considered during 2586 seconds.
5293 make[1]: Leaving directory '/export/home/tools/curl/curl/tests'
5294 status of check cmd=2...

**Comment by grgoffe** [comment]
I don't see how I can attach a full build log. I'm pretty sure you don't want a plain text of the build log and output of make check... so, what should I do if I want to include a file?

Thanks,

George...

**Comment by bagder** [comment]
tldr: I don't think this is a curl bug and I don't see any obvious way to act on this.

## longer version

Rename the logfile to use a `.txt` extension and then you can upload it here.

But, we don't need log that since you already showed the problem. Your OpenSSL build has SSLv3 support still present and enabled and yet its headers says that function is deprecated.

curl will detect if SSLv3 is disabled and not try to use that function. IOW: you're basically asking for this error to happen.

As you have SSLv3 enabled in your SSL library, what do you want curl to do with that?

**Comment by jay** [comment]
Isn't the real problem that he has chosen to error on warnings?

**Comment by bagder** [comment]
Right, and the warnings that are turned into errors are the ones that warn for the deprecated SSLv3 function.

**Comment by grgoffe** [comment]
Hi,

I appreciate your time and energy resolving what I call a UFU (User Foul Up)... So THANKS!

Regards,

George...
